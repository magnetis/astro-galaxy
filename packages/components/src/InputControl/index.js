import { IconInputDash, IconInputPlus } from '@magnetis/astro-galaxy-icons';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import IconGhostButton from '../IconGhostButton';
import InputMasked, { maskTypes } from '../InputMasked';
import { inputSizes } from '../InputText';

const InputControlContainer = styled.div`
  position: relative;
`;

const InputControlButtons = styled.div`
  position: absolute;
  right: ${props => props.theme.space[1]}px;
  top: 5%;
`;

const InputControl = ({
  onIncrement,
  onDecrement,
  incrementLabel,
  decrementLabel,
  buttonProps,
  ...inputProps
}) => {
  const buttonSize =
    inputProps.inputSize === inputSizes.large ? inputSizes.large : inputSizes.medium;
  let buttonVariant = 'ghost.moon';

  if (inputProps.isValidated) {
    buttonVariant = 'ghost.earth';
  }

  if (inputProps.isInvalid) {
    buttonVariant = 'ghost.mars';
  }

  return (
    <InputControlContainer>
      <InputMasked maskType={maskTypes.currency} {...inputProps} displayValidationIcon={false} />
      <InputControlButtons>
        <IconGhostButton
          onClick={onDecrement}
          buttonSize={buttonSize}
          variant={buttonVariant}
          disabled={inputProps.disabled}
          {...buttonProps.decrementButtonProps}>
          <IconInputDash />
        </IconGhostButton>
        <IconGhostButton
          onClick={onIncrement}
          buttonSize={buttonSize}
          variant={buttonVariant}
          disabled={inputProps.disabled}
          {...buttonProps.incrementButtonProps}>
          <IconInputPlus />
        </IconGhostButton>
      </InputControlButtons>
    </InputControlContainer>
  );
};

InputControl.displayName = 'InputControl';

InputControl.defaultProps = {
  buttonProps: {
    incrementButtonProps: {},
    decrementButtonProps: {},
  },
};

InputControl.propTypes = {
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
  buttonProps: PropTypes.shape({
    incrementButtonProps: PropTypes.object,
    decrementButtonProps: PropTypes.object,
  }),
  ...InputMasked.propTypes,
};
export default InputControl;
