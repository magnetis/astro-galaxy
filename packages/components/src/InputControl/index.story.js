import { storiesOf } from '@storybook/react';
import React, { useState } from 'react';
import InputControl from '../InputControl';
import { inputSizes } from '../InputText';

const InputControlSample = ({ ...props }) => {
  const [value, setValue] = useState(0);
  const step = 100;
  const incrementLabel = 'increment';
  const decrementLabel = 'decrement';

  const handleValue = step => setValue(isNaN(value) ? 0 : parseInt(value, 10) + step);

  return (
    <InputControl
      inputId="controlInputId"
      labelId="controlInputLabel"
      labelText="Control input"
      onIncrement={() => handleValue(+step)}
      onDecrement={() => handleValue(-step)}
      buttonProps={{
        incrementButtonProps: {
          'aria-label': incrementLabel,
          title: incrementLabel,
        },
        decrementButtonProps: {
          'aria-label': decrementLabel,
          title: decrementLabel,
        },
      }}
      onChange={event => setValue(event.target.value)}
      thousandSeparator=""
      placeholder="0"
      decimalScale={0}
      value={value}
      {...props}
    />
  );
};

storiesOf('inputs|control inputs', module)
  .add('default', () => <InputControlSample />)
  .add('validated', () => <InputControlSample isValidated />)
  .add('invalid', () => <InputControlSample isInvalid errorMessage="Invalid data" />)
  .add('disabled', () => <InputControlSample disabled={true} />)
  .add('large', () => <InputControlSample inputSize={inputSizes.large} />);
