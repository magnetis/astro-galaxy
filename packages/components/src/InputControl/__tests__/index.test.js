import React from 'react';
import InputControl from '../';

describe('InputControl', () => {
  it('renders correctly', () => {
    const json = rendererCreateWithTheme(
      <InputControl inputId="controlInputId" labelId="controlInputLabel" />
    ).toJSON();

    expect(json).toMatchSnapshot();
  });
});
