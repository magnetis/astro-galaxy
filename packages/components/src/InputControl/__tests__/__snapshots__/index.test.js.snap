// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControl renders correctly 1`] = `
.c6 {
  fill: currentcolor;
  color: #32383c;
}

.c5 {
  cursor: pointer;
  white-space: nowrap;
  text-align: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  overflow: hidden;
  border-width: 2px;
  border-style: solid;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  display: inline-block;
  background-color: transparent;
  color: #597183;
  border-color: transparent;
  padding: 8px 38px;
  font-size: 16px;
  line-height: 28px;
  border-radius: 24px;
  padding: 8px 6px;
}

.c5:disabled {
  cursor: not-allowed;
  background-color: #e9eef2;
  border-color: #e9eef2;
  color: #ffffff;
}

.c5:focus {
  outline: none;
}

.c5 svg {
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  vertical-align: middle;
  margin: -4px 0 -3px;
  width: 32px;
  height: 32px;
}

.c5 svg {
  color: #597183;
}

.c5:disabled {
  background-color: transparent;
  border-color: transparent;
  color: #e9eef2;
}

.c5:disabled svg {
  color: #e9eef2;
}

.c5:hover:not(:disabled) svg,
.c5:focus:not(:disabled) svg {
  color: #8296a4;
}

.c5:hover:not(:disabled),
.c5:focus:not(:disabled) {
  background-color: #f6f7f8;
  border-color: #f6f7f8;
  color: #8296a4;
}

.c5:active:not(:disabled) {
  background-color: #e9eef2;
  border-color: #e9eef2;
  color: #495c69;
}

.c1 {
  border: 0;
  padding: 0;
  position: relative;
}

.c3 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  height: 4em;
  padding: 1em;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #8296a4;
  font-size: 16px;
}

.c2 {
  background-color: #ffffff;
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  color: #32383c;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 25px 16px 5px;
  position: relative;
  width: 100%;
  border-color: #597183;
}

.c2::-webkit-input-placeholder {
  color: transparent;
}

.c2::-moz-placeholder {
  color: transparent;
}

.c2:-ms-input-placeholder {
  color: transparent;
}

.c2::placeholder {
  color: transparent;
}

.c2:focus::-webkit-input-placeholder,
.c2.--no-animate::-webkit-input-placeholder {
  color: #afbec9;
}

.c2:focus::-moz-placeholder,
.c2.--no-animate::-moz-placeholder {
  color: #afbec9;
}

.c2:focus:-ms-input-placeholder,
.c2.--no-animate:-ms-input-placeholder {
  color: #afbec9;
}

.c2:focus::placeholder,
.c2.--no-animate::placeholder {
  color: #afbec9;
}

.c2:focus {
  outline: none;
  border-color: #597183;
}

.c2:focus + label {
  color: #159ce4;
}

.c2:focus,
.c2:not(:placeholder-shown),
.c2.--no-animate {
  border-color: #159ce4;
}

.c2:focus + label,
.c2:not(:placeholder-shown) + label,
.c2.--no-animate + label {
  font-weight: 700;
  -webkit-transform: translate(3px,-6px) scale(0.8);
  -ms-transform: translate(3px,-6px) scale(0.8);
  transform: translate(3px,-6px) scale(0.8);
}

.c2.--no-animate {
  border-color: #597183;
}

.c2.--no-animate:focus {
  border-color: #159ce4;
}

.c2 + label {
  left: 0;
  position: absolute;
  top: 0;
  -webkit-transition: -webkit-transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transition: transform 0.25s,opacity 0.25s ease-in-out;
  transition: transform 0.25s,opacity 0.25s ease-in-out;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c0 {
  position: relative;
}

.c4 {
  position: absolute;
  right: 8px;
  top: 5%;
}

<div
  className="c0"
>
  <fieldset
    className="c1"
  >
    <input
      aria-labelledby="controlInputLabel"
      className="c2  --no-animate"
      id="controlInputId"
      inputMode="numeric"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      placeholder="0,00"
      type="text"
      value=""
    />
    <label
      className="c3"
      htmlFor="controlInputId"
      id="controlInputLabel"
    >
      Standard
    </label>
  </fieldset>
  <div
    className="c4"
  >
    <button
      className="c5"
      display="inline-block"
      onClick={[Function]}
    >
      <svg
        className="c6"
        color="moon900"
        height={24}
        viewBox="0 0 32 32"
        width={24}
      >
        <g
          id="icons/support/less-circle"
        >
          <path
            d="M16.5 29C9.596 29 4 23.404 4 16.5S9.596 4 16.5 4 29 9.596 29 16.5 23.404 29 16.5 29zm0-1.786c5.917 0 10.714-4.797 10.714-10.714 0-5.917-4.797-10.714-10.714-10.714-5.917 0-10.714 4.797-10.714 10.714 0 5.917 4.797 10.714 10.714 10.714zM11.143 16.5h10.714-10.714zm-.893 0c0-.493.4-.893.893-.893h10.714a.893.893 0 0 1 0 1.786H11.143a.893.893 0 0 1-.893-.893z"
            id="Combined-Shape"
          />
        </g>
      </svg>
    </button>
    <button
      className="c5"
      display="inline-block"
      onClick={[Function]}
    >
      <svg
        className="c6"
        color="moon900"
        height={24}
        viewBox="0 0 32 32"
        width={24}
      >
        <g
          id="icons/support/more-circle"
        >
          <path
            d="M16 15.013l3.687-3.687a.77.77 0 0 1 1.088 1.088L17.088 16.1l3.687 3.686a.77.77 0 0 1-1.088 1.088L16 17.188l-3.687 3.687a.77.77 0 0 1-1.088-1.088l3.687-3.686-3.687-3.687a.77.77 0 0 1 1.088-1.088L16 15.013zM28.5 16c0 6.904-5.596 12.5-12.5 12.5S3.5 22.904 3.5 16 9.096 3.5 16 3.5 28.5 9.096 28.5 16zm-2 0c0-5.799-4.701-10.5-10.5-10.5S5.5 10.201 5.5 16 10.201 26.5 16 26.5 26.5 21.799 26.5 16z"
            id="Mask"
            transform="rotate(-45 16 16)"
          />
        </g>
      </svg>
    </button>
  </div>
</div>
`;
